partner_mobile_duplicate_warn/models/res_partner.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Akretion France (http://www.akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    same_mobile_partner_id = fields.Many2one(
+        "res.partner",
+        compute="_compute_same_mobile_partner_id",
+        string="Partner with same mobile",
+        compute_sudo=True,
+    )
+
+    @api.depends("mobile", "company_id")
+    def _compute_same_mobile_partner_id(self):
+        # With phone_validation, the "mobile" field should be
+        # clean in E.164 format, without any start/ending spaces
+        # So we search on the 'mobile' field with '=' !
+        for partner in self:
+            same_mobile_partner_id = False
+            if partner.mobile:
+                domain = [("mobile", "=", partner.mobile)]
+                if partner.company_id:
+                    domain += [
+                        "|",
+                        ("company_id", "=", False),
+                        ("company_id", "=", partner.company_id.id),
+                    ]
+                partner_id = partner._origin.id
+                if partner_id:
+                    domain.append(("id", "!=", partner_id))
+                same_mobile_partner = self.with_context(active_test=False).search(
```

Comment:
Why the `active_test=False`? Maybe unlikely that a phone number might be reused by really another person / company in the same odoo database, but phonenumber can be reused. What is even more likely is that somebody changes his identity (was a single person, now has a company, or the other way around, but keeps the phone). Then not really blocking as only a warning is given.

Reply:
My idea was to be able to identify a partner that was archived, but now needs to be unarchived instead of creating a new partner. But your arguments can also be accepted, so if other people also think we should remove with_context(active_test=False), I'm ok to remove it.

Reply:
@alexis-via Now I understand the use case I am quite satisfied with leaving it as is.